utils/check_repo.py

```diff
@@ -160,6 +160,9 @@
     "ClapAudioModel",
     "ClapAudioModelWithProjection",
     "Blip2ForConditionalGeneration",
+    "Blip2ForImageTextRetrieval",
```

Comment:
This can be removed here and added to the mapping here: https://github.com/huggingface/transformers/blob/e05baad86119efcbf1e2dae64eef0c29f589b07b/src/transformers/models/auto/modeling_auto.py#L1228

Reply:
`Blip2ForImageTextRetrieval` was added to the Zero Shot Image Classification mapping.

- I needed to add `tokenizer_kwargs` to be able to pass `return_token_type_ids`, as blip2 model is currently using bert tokenizer, and token_type_ids is not a valid paramater for blip2.
- use_image_text_matching_head=False is the default value now, to make `Blip2ForImageTextRetrieval` compatible with Zero Shot
- also added test_blip2_model_pt https://github.com/huggingface/transformers/pull/29261/files#diff-ba3a487b37e75e4a7c634b690b000ed4365315743c8f984c6f0f99bf83bc914eR281-R287

Reply:
This still needs to be resolved

Reply:
I just removed `Blip2ForImageTextRetrieval` from `IGNORE_NON_AUTO_CONFIGURED`.
It was already added to `MODEL_FOR_ZERO_SHOT_IMAGE_CLASSIFICATION_MAPPING_NAMES`